encode.go

```diff
@@ -64,7 +80,11 @@ func encodeSQLValue(w io.Writer, arg interface{}) error {
 		// database/driver DefaultParameterConverter. Special handling.
 		value := reflect.ValueOf(arg)
 		if baseKinds[value.Kind()] {
-			return encodeSQLValue(w, asKind(value))
+			asKind := asKind(value)
+			if reflect.DeepEqual(arg, asKind) { // avoid infinite recursion with same arg
+				return fmt.Errorf("unsupported type %T: ConvertValue error: %w", arg, err)
+			}
```

Comment:
this piece would have stopped the infinite recursion which prompted this PR (can see it for oneself by removing the above-added preemptive encoding and running `TestEncodeSQL`)

Reply:
none of the test cases in `TestEncodeSQL` even trigger a single recursive call.
I'm now very dubious of the comment
```go
// We may be in the presence of a type alias not supported by the
// database/driver DefaultParameterConverter. Special handling.
```
and I suspect that the recursive call might only apply for (*)uint(64) cases

Reply:
now that the (*)uint(64) encoding is special-cased before calling `driver.DefaultParameterConverter.ConvertValue`
I'm no longer able even to provoke a recursive call to `encodeSQLValue` even with many type alias test case additions.  That makes me very skeptical of the comment
```go
// We may be in the presence of a type alias not supported by the
// database/driver DefaultParameterConverter. Special handling.
```